src/renderer/components/settings-electron.tsx

```diff
@@ -60,11 +61,47 @@ export const ElectronSettings = observer(
       this.handleStateChange = this.handleStateChange.bind(this);
 
       this.state = {
+        activeVersions: new Set<string>([
+          props.appState.getVersionLockName(props.appState.version),
+        ]),
         isDownloadingAll: false,
         isDeletingAll: false,
       };
     }
 
+    /**
+     * Queries the currently active versions and update the local state.
+     *
+     * This currently gives a warning/error in development mode when the ElectronSettings component is unmounted
+     * ("Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak
+     * in your application"). This is a false positive, the warning has been removed in React 18+ (see
+     * https://github.com/facebook/react/pull/22114).
+     *
+     * @TODO upgrade to React 18
+     */
+    updateActiveVersions = () => {
+      this.props.appState
+        .getActiveVersions()
+        .then((activeVersions) =>
+          this.setState({ ...this.state, activeVersions }),
+        )
+        .catch((err) => {
+          console.error(
+            'Error updating the currently active Electron versions:',
+          );
+          console.error(err);
+        });
+    };
+
+    public componentDidMount() {
+      this.updateActiveVersions();
+    }
+
+    // Fired when other windows change their active Electron version
+    public componentDidUpdate() {
+      this.updateActiveVersions();
+    }
```

Comment:
I'm not quite understanding the mechanism by which this works - what causes another window changing its active Electron version to trigger `componentDidUpdate` here? I can see it's working based on your video, I just don't understand it. 🤔

Reply:
@erikian, poked around with it locally and figured it out. Since `updateActiveVersions` unconditionally modifies the state, which triggers a component update, this is effectively creating an infinite loop on the component updating, which is why it appears to be working as intended.

Reply:
What if we extended the broadcast mechanism in #1391 to add a message (`activeVersionsChanged`, for example), move `activeVersions` to `AppState`, and then in response to that message we update `AppState.activeVersions`? The broadcast message would have no payload, it would just cause `AppState` to update `activeVersions`, which would then cause an update to the `ElectronSettings` component.

Reply:
> Since updateActiveVersions unconditionally modifies the state, which triggers a component update, this is effectively creating an infinite loop on the component updating, which is why it appears to be working as intended.

My bad big time 🤦‍♂️ I thought this component was re-rendering because of the `storage` event that is fired when the other window selects a version.

> What if we extended the broadcast mechanism in https://github.com/electron/fiddle/pull/1391 to add a message (activeVersionsChanged, for example), move activeVersions to AppState, and then in response to that message we update AppState.activeVersions? The broadcast message would have no payload, it would just cause AppState to update activeVersions, which would then cause an update to the ElectronSettings component.

Sounds good to me.